pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -478,6 +487,46 @@ func validatingHasAcceptedAdmissionReviewVersions(webhooks []admissionregistrati
 	return true
 }
 
+// mutatingChangedMatchConditionExpressions returns true if any new expressions are added
+func mutatingChangedMatchConditionExpressions(new, old []admissionregistration.MutatingWebhook) bool {
+	expressions := sets.NewString()
+	for _, hook := range old {
+		for _, oldMatchCondition := range hook.MatchConditions {
+			expressions.Insert(oldMatchCondition.Expression)
+		}
+	}
+
+	for _, hook := range new {
+		for _, newMatchCondition := range hook.MatchConditions {
+			if !expressions.Has(newMatchCondition.Expression) {
+				return true
+			}
+		}
+	}
+
+	return false
```

Comment:
If you go with my above suggestion to skip all match conditions validation if they are unchanged, I think this can be simplified to just `return !reflect.DeepEqual(new, old)`.

Otherwise, I think you can just assume the order doesn't change, and simplify to:
```go
if len(old) != len(new) {
  return true
}
for i := range old {
  if new[i].Expression != old[i].Expression {
     return true
   }
}
return false 
```

Reply:
Yeah I guess the advantage of this way is just thinking from a perspective of recompiling CEL expressions. We would only need to run the compile test for new expressions that weren't already compiled. That's what I was going for at least but again I'm happy to make any updates. `return !reflect.DeepEqual(new, old)` would be nice and clean

Reply:
so like for example, if the same match condition expressions were reordered or renamed. Or we removed some expressions. We wouldn't need to recompile.